ioservice/cob_foms.c

```diff
@@ -480,17 +480,20 @@ static int cob_setattr_fom_tick(struct m0_fom *fom)
 static bool cob_pool_version_mismatch(const struct m0_fom *fom)
 {
 	int                       rc;
-	struct m0_cob            *cob;
+	struct m0_cob            *cob = NULL;
 	struct m0_fop_cob_common *common;
+	bool                      ret;
 
 	common = m0_cobfop_common_get(fom->fo_fop);
 	rc = cob_locate(fom, &cob);
 	if (rc == 0 && cob != NULL) {
 		M0_LOG(M0_DEBUG, "cob pver"FID_F", common pver"FID_F,
 				FID_P(&cob->co_nsrec.cnr_pver),
 				FID_P(&common->c_body.b_pver));
-		return !m0_fid_eq(&cob->co_nsrec.cnr_pver,
-				  &common->c_body.b_pver);
+		ret = !m0_fid_eq(&cob->co_nsrec.cnr_pver,
+				 &common->c_body.b_pver);
+		m0_cob_put(cob);
```

Comment:
Please add comments for interface `m0_cob_locate()` to say that:
If this function returns success, the result cob must be released by calling `m0_cob_put()`.

Please add this in the cob/cob.[ch] for the declaration and/or definition.

Reply:
Also, the comments of `m0_cob_locate()` need updates. It has "@param tx    db transaction to be used" which is not there.

Reply:
Done.